plugins/processors/k8sdecorator/stores/utils.go

```diff
@@ -30,11 +31,12 @@ func createContainerKeyFromMetric(tags map[string]string) string {
 	return k8sutil.CreateContainerKey(namespace, podName, containerName)
 }
 
-const (
-	// kubeAllowedStringAlphaNums holds the characters allowed in replicaset names from as parent deployment
+var (
+	// deploymentUnexpectedRegEx holds the characters allowed in replicaset names from as parent deployment
 	// https://github.com/kubernetes/apimachinery/blob/master/pkg/util/rand/rand.go#L83
-	kubeAllowedStringAlphaNums = "bcdfghjklmnpqrstvwxz2456789"
-	cronJobAllowedString       = "0123456789"
+	deploymentUnexpectedRegExp = regexp.MustCompile(`[^b-hj-np-tv-xz-z2-24-9]+`)
```

Comment:
I think this naming is throwing me off a bit.  The variable name makes it seem like the characters in the regex are not allowed and all other characters are allowed, however, I think the regex is defining only the allowed characters.

Reply:
Yes, the regexp is for checking un-allowed character in their name. In this way it would be easier to check for allowed character though.

Reply:
I agree with @jhnlsn. The name of the variable is contradictory. It inverts the original one (this applies to both variables).

I think the name of the variable should be changed to reflect it's usage. Even in the updated `deploymentUnallowed`, it is directly contradictory to the original variable name, `kubeAllowed`, but matches the same characters.

My suggestion is to reuse what was already there:
```go
var (
    kubeAllowedChars = regexp.MustCompile()
    cronJobAllowedChars = regexp.MustCompile()
)
```

Reply:
Looked over the old code, and I still stand by my opinion on this. The way that the strings were used previously, the code iterated over the input string, and checked if there were any chars that were not contained in the strings above. Switching from that to a regular expression makes sense. 

That being said, the naming is confusing because what we want to ensure is that the name of the cron job **matches** the two patterns.

Reply:
Yes. I agrees that it needs to ```matches``` the two patterns. However, the two patterns are for unallowed character (meaning name does not allow these characters). Therefore, I am still unconvincing that we should use ```allowedChars```.

Reply:
I think what we should do is flip the regular expression. I see why I'm confused now.

The original code would check:
```go
if !stringInRuneset(suffix, "bcdfghjklmnpqrstvwxz2456789")
```

the `stringInRuneset()` function iterates over `suffix`, and then if a character in `suffix` is NOT in the string `"bcdfghjklmnpqrstvwxz2456789"`, then it returns false, indicating that there was a character that should not have been there. For example, `stringInRuneset("bghk2789?", "bcdfghjklmnpqrstvwxz2456789")` would return `false` because ? is not an allowable character.

---
What you're changing it to is using a regular expression.
With the regex you defined:
```go
deploymentUnallowedRegExp = regexp.MustCompile(`[^b-hj-np-tv-xz-z2-24-9]+`)
```
I finally noticed the `^` for negativing it. 

This is what makes the variable and usage confusing. We should avoid using negative wording as it can be confusing. We should change it to affirmative wording, like what I suggested, and then flip the regular expression like:
```go
kubeAllowedChars = regexp.MustCompile(`[b-hj-np-tv-xz-z2-24-9]+`)
```

Reply:
For my opinion, flipping with ! outside is different from flipping with ! inside the regex though. For example, the `2a` would return false; however, it expected to be return true.

Reply:
Why is it expected to return true? We control whether it returns true or false. 

The original code was as follows, and expected to return `false`, which gets negated.
```go
if !stringInRuneset(suffix, kubeAllowedStringAlphaNums) {
```
Your change inverts the if statement, so it drops the negation:
```go
if deploymentUnallowedRegExp.MatchString(suffix) {
```

If we change the regex to `regexp.MustCompile(`[b-hj-np-tv-xz-z2-24-9]+`)`, this checks if the input string contains only those characters.

So for an input suffix like `bjjk24789`, doing `MatchString()` on this with the above regex will return `true`. This indicates that the suffix is **valid**. As such, if we update the conditionals to expect false (which is what it was before your change), we maintain the correct functionality.

```go
if kubeAllowedChars.MatchString(suffix) {
```

Reply:
Yes we can control whether it returns true or false. However, my main point is the conditions with the RegEx. For example, checking whether not the number and whether it has the character besides number is different. 
To be more specific about that, 
```
if !kubeAllowedChars.MatchString(suffix) {
   return ""
}
```
and 
```
if kubeUnallowedChars.MatchString(suffix){
  return ""
}
```
is different though. Does the first case pass the "2ab", simply no . However, the second case passes.

Reply:
See my discussion on this [here](https://github.com/aws/amazon-cloudwatch-agent/pull/345#discussion_r795713916)